packages/forms/examples/advanced.md

```diff
@@ -0,0 +1,162 @@
+    : 'valid entry';
+
+const getTextValidation = (value, minimum = 10) =>
+  value.length === 0 ? 'error' : value.length < minimum ? 'warning' : 'success';
+
+<>
+  <Field>
+    <Label>Controlled text input</Label>
+    <Hint>Entry must contain at least 10 characters</Hint>
+    <Input
+      value={state.inputValue}
+      onChange={event => setState({ inputValue: event.target.value })}
+    />
+    <Message validation={getTextValidation(state.inputValue)}>
+      {getTextMessage(state.inputValue)}
+    </Message>
+  </Field>
+  <Field className="u-mt">
+    <Label>Controlled textarea</Label>
+    <Hint>Entry must contain at least 20 characters</Hint>
+    <Textarea
+      value={state.textareaValue}
+      onChange={event => setState({ textareaValue: event.target.value })}
+    />
+    <Message validation={getTextValidation(state.textareaValue, 20)}>
+      {getTextMessage(state.textareaValue, 20)}
+    </Message>
+  </Field>
+  <Field className="u-mt">
+    <Label>Controlled media input</Label>
+    <Hint>Entry must contain at least 5 characters</Hint>
+    <MediaInput
+      value={state.mediaInputValue}
+      onChange={event => setState({ mediaInputValue: event.target.value })}
+      start={<StartIcon />}
+      end={<EndIcon />}
+    />
+    <Message validation={getTextValidation(state.mediaInputValue, 5)}>
+      {getTextMessage(state.mediaInputValue, 5)}
+    </Message>
+  </Field>
+  <div className="u-mt" role="group" aria-label="controlled radio">
+    <Field>
+      <Label>Controlled radio</Label>
+    </Field>
+    <Field>
+      <Radio
+        name="controlled-example"
+        value="error"
+        checked={state.radioValue === 'error'}
+        onChange={event => setState({ radioValue: event.target.value })}
+      >
+        <Label isRegular>Error</Label>
+      </Radio>
+    </Field>
+    <Field>
+      <Radio
+        name="controlled-example"
+        value="warning"
+        checked={state.radioValue === 'warning'}
+        onChange={event => setState({ radioValue: event.target.value })}
+      >
+        <Label isRegular>Warning</Label>
+      </Radio>
+    </Field>
+    <Field>
+      <Radio
+        name="controlled-example"
+        value="success"
+        checked={state.radioValue === 'success'}
+        onChange={event => setState({ radioValue: event.target.value })}
+      >
+        <Label isRegular>Success</Label>
+      </Radio>
+      <Message validation={state.radioValue}>{getRadioMessage(state.radioValue)}</Message>
+    </Field>
+  </div>
+  <Field className="u-mt">
+    <Checkbox
+      checked={state.checkboxChecked}
+      indeterminate={state.checkboxIndeterminate}
+      onChange={event =>
+        setState({
+          checkboxChecked: event.target.checked,
+          checkboxIndeterminate: event.nativeEvent.shiftKey
+        })
+      }
+    >
+      <Label>Controlled checkbox</Label>
+      <Hint>
+        Must be checked (hold shift key for <Code>indeterminate</Code>)
```

Comment:
The indeterminate checkbox works for me on latest Chrome, Safari, and IE11, but it doesn't seem to work on Firefox (68.0.2).

Reply:
You found a cool issue with FF: https://bugzilla.mozilla.org/show_bug.cgi?id=559506. I'm not going to sub out the key, as there are various issues with the other options (alt, cmd, ctrl) – so let's leave this as a minor quirk limited to the example.

Reply:
@jzempel thanks for looking into it, much appreciated!